core/search/disjunction.hpp

```diff
@@ -284,6 +462,11 @@ class small_disjunction : public doc_iterator_base, score_ctx {
     }
 
     doc_ = min;
+
+    if /*constexpr*/ (std::is_same<Adapter, position_score_iterator_adapter<DocIterator>>::value) {
+      this->update_attribute_range(doc_.value, itrs_);
+    }
+
```

Comment:
let's avoid having this logic as a part of next/seek call, use attribute+visitor instead